File: ios/RCTMGL/RCTMGLImageSourceManager.m
//
//  RCTMGLImageSourceManager.m
//  RCTMGL
//
//  Created by Nick Italiano on 11/29/17.
//  Copyright © 2017 Mapbox Inc. All rights reserved.
//

#import "RCTMGLImageSourceManager.h"
#import "RCTMGLImageSource.h"

@implementation RCTMGLImageSourceManager

RCT_EXPORT_MODULE()

RCT_EXPORT_VIEW_PROPERTY(id, NSString)
RCT_EXPORT_VIEW_PROPERTY(url, NSString)
RCT_EXPORT_VIEW_PROPERTY(coordinates, NSArray)

- (UIView*)view
{
    return [RCTMGLImageSource new];
}

@end

File: ios/RCTMGL-v10/RCTMGLImageSourceManager.swift
@objc(RCTMGLImageSourceManager)
class RCTMGLImageSourceManager: RCTViewManager {
  @objc
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }

  @objc override func view() -> UIView {
    return RCTMGLImageSource()
  }
}
